### PR TITLE
Add async_generator

### LINKED ISSUE #38: Latest release requires async_generator for Python 3.6
Hello, the dependency list needs to be updated to include `async_generator` for Python 3.6:
https://github.com/encode/httpx/blob/9b17671f155aa857fb52911d94357147ba2a3ea4/setup.py#L63

This means that the package can no longer be `noarch`.

Happy to make a PR if you're too busy. Let me know what you think.

Best wishes,
Richard

### PR DESCRIPTION
**Pull request by PrimozGodec**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Fixes https://github.com/conda-forge/httpx-feedstock/issues/38

### CODE REVIEW

**Comment by PrimozGodec** [comment]
@conda-forge-admin, please rerender

**Comment by PrimozGodec** [comment]
@conda-forge-admin, please rerender

**Comment by PrimozGodec** [comment]
@conda-forge-admin, please restart ci

File: recipe/meta.yaml [hunk 1]
```diff
@@ -10,21 +10,22 @@ source:
   sha256: 0a2651dd2b9d7662c70d12ada5c290abcf57373b9633515fe4baa9f62566086f
 
 build:
-  noarch: python
-  number: 0
+  number: 1
   script: {{ PYTHON }} -m pip install . -vv
+  skip: true  # [py<36]
```
Comment (hoechenberger) [inline recipe/meta.yaml]: You can remove this line as conda-forge doesn't support anything older than Python 3.6 anymore.

Otherwise looks good to me!

Comment (PrimozGodec) [reply recipe/meta.yaml]: You are right. I removed the constraint and will merge after the test passes. Thanks

**Comment by PrimozGodec** [comment]
@conda-forge-admin, please rerender

**Comment by hoechenberger** [comment]
Thanks @PrimozGodec!

**Pull request merged**